### PR TITLE
Limit group separator to one single character in Prefs->Entry Keyword Separator

### PR DESCRIPTION
**Pull request by colinldbd**
Fixed issue  [#177 Limit group separator](https://github.com/koppor/jabref/issues/177)  where Prefs dialog for groups should be adjusted that it is only possible to insert a single character as delimiter.

In EntryTab.java I used Text Formatter to limit the input to TextField Keyword Separator to only 1 character.

The original TextField for Keyword Separator in issue [#177 Limit group separator](https://github.com/koppor/jabref/issues/177):
image

However, the position of Keyword Separator has been moved to Preference->Entry, it's no longer in Preference->Group:
image

The current keyword separator is here:
image

Before, you can input string for whatever length into the keyword separator 
image

But after the change, there's only one single character allowed:
image

I managed to create a TextFormatter for the TextField keywordSeparatr and limited the input string to one single character.

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,11 +7,11 @@ In case, there is no issue present, the pull request implementing the feature is
 
 Note that this project **does not** adhere to [Semantic Versioning](https://semver.org/).
 
-## [Unreleased]
+## [Unreleased] – 2023-10-22
```
Comment (Siedlerchr) [inline CHANGELOG.md:10]: don't worry about this, there is another fix for this

Comment (colinldbd) [reply CHANGELOG.md:10]: Thanks for the reminder, I'm just worried why I can't pass the Markdown style check, but I guess for now it I don't need to do anything about it?

Comment (colinldbd) [reply CHANGELOG.md:10]: > don't worry about this, there is another fix for this

Thanks for the reminder, I'm just worried why I can't pass the Markdown style check, but I guess for now it I don't need to do anything about it?

**Comment by colinldbd** [comment]
I'm so sorry about this but I just realized that my issue belongs to [koppor](https://github.com/koppor)
/
[jabref](https://github.com/koppor/jabref) instead of 
https://github.com/JabRef
/
[jabref](https://github.com/JabRef/jabref)

Would that be a problem, I didn't realize there's 2 repo for this project.

You can close this pull-request if you want.

**Comment by Siedlerchr** [comment]
No, this is correct. Fixes go here, just change the link

**Comment by colinldbd** [comment]
> No, this is correct. Fixes go here, just change the link

Thanks for the reply! Please feel free to review my PR and I'll make the revise if needed.

File: CHANGELOG.md (hunk 1 above)
Comment (koppor) [reply CHANGELOG.md:10]: You need to worry a bit. The issue is quite clear:

     10:1  error  Unexpected additional part: ' – 2023-10-22'  entry-for-every-versions

An unreleased version has no release date - but you added one ^^. Just remove it again.

Please also merge latest `main`, which improved the CHANGELOG.md check - https://github.com/JabRef/jabref/pull/10537

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 ### Added
 
 ### Changed
-
+- WE changed the TextField for KeywordSeparator to only accepting 1 single character by using a TextFormatter in EntryTab.java so that Prefs dialog for groups should be adjusted that it is only possible to insert a single character as delimiter.[#177](https://github.com/koppor/jabref/issues/177)
```
Comment (koppor) [inline CHANGELOG.md]: I can tell you what the linter says: Missing empty lines.

See https://github.com/DavidAnson/markdownlint/blob/main/doc/md022.md for details.

**Comment by koppor** [comment]
There was a strange diff, therefore, [`1e5af74` (#10543)](https://github.com/JabRef/jabref/pull/10543/commits/1e5af749215f4a5d2649295760410b93cbe45b65) and [`6c06d90` (#10543)](https://github.com/JabRef/jabref/pull/10543/commits/6c06d90cc7ab35f7eb6b4ec137ce62fe5d4d20ee) have been introduced.

**Comment by koppor** [comment]
I updated the CHANGELOG.md to be more end-user-oriented.

**Review by calixtus — APPROVED** [review]

**Comment by koppor** [comment]
@colinldbd We refined the overall commits a bit (there were strange updates in) and added the thing to the merge queue.

**Comment by ThiloteE** [comment]
@colinldbd 
To ease organizational workflows I have linked the pull-request to the issue with syntax as described in https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

> <h2 id="linking-a-pull-request-to-an-issue-using-a-keyword">Linking a pull request to an issue using a keyword</h2>
> <p>You can link a pull request to an issue by using a supported keyword 
> in the pull request's description or in a commit message. The pull 
> request <strong>must be</strong> on the default branch.</p>
> <ul>
> <li>close</li>
> <li>closes</li>
> <li>closed</li>
> <li>fix</li>
> <li>fixes</li>
> <li>fixed</li>
> <li>resolve</li>
> <li>resolves</li>
> <li>resolved</li>
> </ul>
> <p>If you use a keyword to reference a pull request comment in another 
> pull request, the pull requests will be linked. Merging the referencing 
> pull request also closes the referenced pull request.</p>
> <p>The syntax for closing keywords depends on whether the issue is in the same repository as the pull request.</p>


### Example:
- ✅ `Fix #xyz` links pull-request to issue. Merging the PR will close the Issue.
- ✅ `Fix https://github.com/JabRef/jabref/issues/xyz` links pull-request to issue. Merging the PR will close the Issue.
- ✅ `Fix https://github.com/Koppor/jabref/issues/xyz` links pull-request to issue. Merging the PR will close the Issue.
- ❌ `Fix [#xyz](https://github.com/JabRef/jabref/issues/xyz)` links pull-request to issue. Merging the PR will **NOT** close the Issue.

**Pull request merged**